converters/toZigbee.js

```diff
@@ -4849,6 +4849,63 @@ const converters = {
             await entity.read('schneiderSpecificPilotMode', ['pilotMode'], {manufacturerCode: 0x105e});
         },
     },
+    schneider_temperature_measured_value: {
+        key: ['temperature_measured_value'],
+        convertSet: async (entity, key, value, meta) => {
+            await entity.report('msTemperatureMeasurement', {'measuredValue':Math.round(value*100)});
+        },
+    },
+    schneider_thermostat_system_mode: {
+        key: ['system_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            let systemMode = utils.getKey(constants.thermostatSystemModes, value, undefined, Number);
+            if (systemMode === undefined) {
+                systemMode = utils.getKey(legacy.thermostatSystemModes, value, value, Number);
+            }
+            globalStore.putValue(entity, 'systemMode', systemMode);
```

Comment:
Is the only reason for this to buffer the commands until the device awakes?

Reply:
These attributes are read by the device, they cannot be written.

Reply:
What if we just send a readReponse here? (without the device requested for one). Does that also work?

Reply:
When the device is asleep, it doesn't acknowledge any frame.

Also, spamming readResponses to the device does not seem to work either, even when awake. The device seems to correlate the responses to its requests somehow (with sequence number maybe?)

Reply:
I've updated herdsman to allow read response to all attributes. Can you check if it works?  (update to the latest dev branch first). After that:

- Remove the read response code from the `onEvent` handler
- Replace `globalStore.putValue(entity, 'systemMode', systemMode);` with `endpoint.saveClusterAttributeKeyValue('hvacThermostat', {systemMode: systemMode})`

This has the benefit that these attributes are persisted when z2m is restarted.

Reply:
1) Do I still need the hacked firmware/modified startZnp of herdsman for this to work?

2) I'm not sure how to update herdsman to the latest dev version within my zigbee2mqtt npm environment.

Reply:
1. Firmware yes, modified startZnp no
2. Just follow https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html#bare-metal again

Reply:
Tested and it works, I've updated the code accordingly.
I still need to use globalStore for pending writes to keypad_lockout and for handling of schneider_ui_action where I need to know if screen is awake or not.

Reply:
I'm sorry to come back on that issue, but I installed my production environnement and the the thermostat no longer take the readResponses into account. I think I messed-up when I tested with the hacked firmware a while ago.

After some troubleshooting, the readResponses are sent from coordinator endpoint 3, and the sent attributes are correct, however the profile is incorrect: should be 0x0104 for Home Automation, and is 0x0105.

How can I set the profile for the read responses?